File: luaScripts/test_messaging.lua
package.cpath = package.cpath .. ";../cmake-build-debug/?.dylib"

local messaging = require("messaging")

messaging.subscribe("sensor.data")

local result = messaging.publish{
	topic = "sensor.data",
	payload = "NewYork:0",
	source = "messaging lua test"
}

print("Message ID: ", result.messageId)
print("Timestamp: ", result.timestamp)

File: pythonScripts/test_messaging.py
import sys
sys.path.append("../cmake-build-debug")

import messaging_py as msg

print("messaging_py module loaded")

messaging = msg.Messaging()

def on_sensor(message):
    print("Received message")
    print("Topic:", message.topic)
    print("Payload:", message.payload)

messaging.subscribe("sensor.data", on_sensor)

m = msg.Message()
m.topic = "sensor.data"
m.payload = "target detected"
m.timestamp = 10
m.source = "sensor"

published = messaging.publish(m)
